bitnami/argo-workflows/templates/_helpers.tpl

```diff
@@ -0,0 +1,243 @@
+    {{ default (printf "%s-server" (include "common.names.fullname" .)) .Values.server.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.server.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the name of the controller service account to use
+*/}}
+{{- define "argo-workflows.controller.serviceAccountName" -}}
+{{- if .Values.controller.serviceAccount.create -}}
+    {{ default (printf "%s-controller" (include "common.names.fullname" .)) .Values.controller.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.controller.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create the name of the workflows service account to use
+*/}}
+{{- define "argo-workflows.workflows.serviceAccountName" -}}
+{{- if .Values.workflows.serviceAccount.create -}}
+    {{ default (printf "%s-workflows" (include "common.names.fullname" .)) .Values.workflows.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.workflows.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "argo-workflows.imagePullSecrets" -}}
+{{- include "common.images.pullSecrets" (dict "images" (list .Values.server.image .Values.controller.image) "global" .Values.global) -}}
+{{- end -}}
+
+{{/*
+Return the proper configmap for the controller
+*/}}
+{{- define "argo-workflows.controller.configMapName" -}}
+{{- if .Values.controller.existingConfigMap }}
+{{- .Values.controller.existingConfigMap -}}
+{{- else -}}
+{{- include "argo-workflows.controller.fullname" . -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if persistence is enabled
+*/}}
+{{- define "argo-workflows.controller.persistence.enabled" -}}
+{{- if or .Values.postgresql.enabled .Values.mysql.enabled .Values.externalDatabase.enabled -}}
+{{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper database username
+*/}}
+{{- define "argo-workflows.controller.database.username" -}}
+{{- if .Values.postgresql.enabled -}}
+{{- .Values.postgresql.postgresqlUsername -}}
+{{- end -}}
+{{- if .Values.mysql.enabled -}}
+{{- .Values.mysql.auth.username -}}
+{{- end -}}
+{{- if .Values.externalDatabase.enabled -}}
+{{- .Values.externalDatabase.username -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper database username secret name
+*/}}
+{{- define "argo-workflows.controller.database.username.secret" -}}
+{{- printf "%s-%s" (include "argo-workflows.controller.fullname" .) "database" -}}
+{{- end -}}
+
+{{/*
+Return the proper database password secret
+*/}}
+{{- define "argo-workflows.controller.database.password.secret" -}}
+{{- if .Values.postgresql.enabled -}}
+{{- include "argo-workflows.postgresql.fullname" . -}}
+{{- end -}}
+{{- if .Values.mysql.enabled -}}
+{{- include "argo-workflows.mysql.fullname" . -}}
+{{- end -}}
+{{- if .Values.externalDatabase.enabled -}}
+{{- if .Values.externalDatabase.existingSecret -}}
+{{- .Values.externalDatabase.existingSecret -}}
```

Comment:
Allow rendering for this existingSecret, users may want to use an umbrella chart where they create the secret and then use Argo as a subchart (this practice is done by charts like Gitlab)